src/vfio_device.rs

```diff
@@ -58,12 +58,8 @@ pub type Result<T> = std::result::Result<T, VfioError>;
 impl fmt::Display for VfioError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
-            VfioError::OpenContainer(e) => {
-                write!(f, "failed to open /dev/vfio/vfio container: {}", e)
-            }
-            VfioError::OpenGroup(e, ref p) => {
-                write!(f, "failed to open /dev/vfio/{} group: {}", p, e)
-            }
+            VfioError::OpenContainer(_e) => write!(f, "failed to open /dev/vfio/vfio container"),
```

Comment:
Why not:
```Rust
VfioError::OpenContainer(e) => write!(f, "failed to open /dev/vfio/vfio container: {}", e.source().unwrap()),
```

Reply:
There's discussion about this topic here: https://github.com/rust-vmm/event-manager/pull/28#pullrequestreview-448709021

Basically we hope to follow the standard std::error::Error design pattern.

Reply:
So from the VMM code, if I get a VFIO error, I would have to implement the following pattern to get the whole chain:

```Rust
error!("VFIO error {:?}", e.source().unwrap())
```

Or use the `anyhow` crate to handle it for me?

Reply:
anyhow is recommended, and cloud hypervisor already uses anyhow anyway.
It's really hard to compose the error message when the error has a deep chain.